pyproject.toml

```diff
@@ -1,10 +1,259 @@
+dev-release = [
+    "towncrier ~= 22.8",
+    "pydoctor ~= 22.9.0",
+    "sphinx-rtd-theme ~= 0.5",
+    "readthedocs-sphinx-ext ~= 2.1",
+    "sphinx >= 4.1.2, <6",
+]
+dev_release = [
+    "twisted[dev-release]",
+]
+
+# All the extra tools used to help with the development process.
+dev = [
+    "twisted[dev-release]",
+    "pyflakes ~= 2.2",
+    # TODO: support python-subunit in py3.10 https://twistedmatrix.com/trac/ticket/10115
+    "python-subunit ~= 1.4; python_version < '3.10'",
+    "twistedchecker ~= 0.7",
+    "coverage >= 6b1, <7",
+]
+
+tls = [
+    "pyopenssl >= 21.0.0",
+    # service_identity 18.1.0 added support for validating IP addresses in
+    # certificate subjectAltNames
+    "service_identity >= 18.1.0",
+    "idna >= 2.4",
+]
+
+conch = [
+    "pyasn1",
+    "cryptography >= 2.6",
+    "appdirs >= 1.4.0",
+    "bcrypt >= 3.0.0",
+]
+
+conch-nacl = [
+    "twisted[conch]",
+    # Used to support Ed25519 keys on systems with OpenSSL < 1.1.1b
+    "PyNaCl",
+]
+conch_nacl = [
+    "twisted[conch-nacl]",
+]
+
+serial = [
+    "pyserial >= 3.0",
+    "pywin32 != 226; platform_system == 'Windows'",
+]
+
+http2 = [
+    "h2 >= 3.0, < 5.0",
+    "priority >= 1.1.0, < 2.0",
+]
+
+contextvars = [
+    "contextvars >= 2.4, < 3; python_version < '3.7'",
+]
+
+all-non-platform = [
+    "twisted[test,tls,conch,serial,http2,contextvars]",
+]
+all_non_platform = [
+    "twisted[all-non-platform]",
+]
+
+macos-platform = [
+    "twisted[all-non-platform]",
+    "pyobjc-core",
+    "pyobjc-framework-CFNetwork",
+    "pyobjc-framework-Cocoa",
+]
+macos_platform = [
+    "twisted[macos-platform]",
+]
+
+windows-platform = [
+    "twisted[all-non-platform]",
+    "pywin32 != 226",
+]
+windows_platform = [
+    "twisted[windows-platform]",
+]
+
+osx-platform = [
+    "twisted[macos-platform]",
+]
+osx_platform = [
+    "twisted[osx-platform]",
+]
+
+gtk-platform = [
+    "twisted[all-non-platform]",
+    "pygobject",
+]
+gtk_platform = [
+    "twisted[gtk-platform]",
+]
+
+mypy = [
+    "twisted[dev,all-non-platform,conch-nacl]",
+    "mypy==0.930",
+    "mypy-zope==0.3.4",
+    "types-setuptools",
+    "types-pyOpenSSL",
+]
+
+[project.scripts]
+cftp = "twisted.conch.scripts.cftp:run"
+ckeygen = "twisted.conch.scripts.ckeygen:run"
+conch = "twisted.conch.scripts.conch:run"
+mailmail = "twisted.mail.scripts.mailmail:run"
+pyhtmlizer = "twisted.scripts.htmlizer:run"
+tkconch = "twisted.conch.scripts.tkconch:run"
+trial = "twisted.scripts.trial:run"
+twist = "twisted.application.twist._twist:Twist.main"
+twistd = "twisted.scripts.twistd:run"
+
+[project.urls]
+Changelog = "https://github.com/twisted/twisted/blob/HEAD/NEWS.rst"
+Documentation = "https://docs.twistedmatrix.com/"
+Homepage = "https://twistedmatrix.com/"
+Issues = "https://twistedmatrix.com/trac/report"
+Source = "https://github.com/twisted/twisted"
+Twitter = "https://twitter.com/twistedmatrix"
+
+[tool.hatch.metadata]
+allow-ambiguous-features = true
```

Comment:
Can we have a comment for this usage ?
Why we need it here?

Maybe this is correct 

```suggestion
# This is here to enable backward compatible extra dependencies
# that are defined using underscores.
allow-ambiguous-features = true
```